test/e2e/utils/helper.go

```diff
@@ -273,10 +274,53 @@ func CreateResourcesForWebHookE2E(ctx context.Context, hubCluster *framework.Clu
 	gomega.Eventually(func() error {
 		return hubCluster.KubeClient.Create(ctx, &rb)
 	}, PollTimeout, PollInterval).Should(gomega.Succeed(), "failed to create role binding %s for webhook E2E", rb.Name)
+
+	// Creating this MC for IMC E2E, this MC will fail to join since it's name is not configured to be recognized by the member agent.
+	// But it serves its purpose for the tests.
+	identity := rbacv1.Subject{
+		Name:      "test-user",
+		Kind:      "ServiceAccount",
+		Namespace: utils.FleetSystemNamespace,
+	}
+	mc := &fleetv1alpha1.MemberCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-mc",
+		},
+		Spec: fleetv1alpha1.MemberClusterSpec{
+			Identity:               identity,
+			State:                  fleetv1alpha1.ClusterStateJoin,
+			HeartbeatPeriodSeconds: 60,
+		},
+	}
+	gomega.Eventually(func() error {
+		return hubCluster.KubeClient.Create(ctx, mc)
+	}, PollTimeout, PollInterval).Should(gomega.Succeed(), "Failed to wait for member cluster %s to be created in %s cluster", mc.Name, hubCluster.ClusterName)
+
+	imc := &fleetv1alpha1.InternalMemberCluster{
```

Comment:
where is this used ?

Reply:
https://github.com/Azure/fleet/blob/7cae41bbb51c01af76cba390cfe38e483453af98/test/e2e/webhook_test.go#L710 in the E2E tests

Reply:
I am a bit confused, the line you pasted declared a new variable.